internal/bminventory/inventory.go

```diff
@@ -437,6 +437,10 @@ func (b *bareMetalInventory) UpdateDiscoveryIgnition(ctx context.Context, params
 		return installer.NewUpdateDiscoveryIgnitionInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
 
+	if err := b.objectHandler.DeleteObject(ctx, getImageName(*c.ID)); err != nil {
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+
```

Comment:
Please emit an event, something like:
b.eventsHandler.AddEvent(ctx, *c.ID, nil, models.EventSeverityInfo, "Deleted image from backend because its ignition was updated. The image may be generated again at any time.", time.Now())

Reply:
Sure, do you also think one is necessary for the error case?

Also I'm not sure we can tell if something was actually deleted so we'll get this event every time we edit the ignition even if the image hasn't actually been generated yet.

Reply:
If we think that will be too much noise I would have to add an additional call to see if the object exists first which wouldn't otherwise be necessary.

Reply:
Yes I event for error is a good idea.  Maybe instead of checking for existence, change DeleteObject to optionally return an error if the object doesn't exist?

Reply:
> Maybe instead of checking for existence, change DeleteObject to optionally return an error if the object doesn't exist?

I don't really want an error .. I just want to know if any real work was done. Maybe an additional return value to indicate if something was actually deleted?

Reply:
Sure